src/pygeosx/unitTests/testWaveEquationAdjoint.py

```diff
@@ -0,0 +1,279 @@
+    else:
+        tmp1=np.zeros((n1,n2))
+        for i in range(1,nRank):
+            comm.Recv(tmp1,i,1)
+            for j in range(0,n2):
+                if np.abs(np.min(tmp[:,j]))<0.0000001 and np.abs(np.max(tmp[:,j]))<0.00000001: 
+                    tmp[:,j]+=tmp1[:,j]
+
+    comm.Barrier()
+    return tmp
+
+def print_group(group, indent=0):
+    print("{}{}".format(" " * indent, group))
+
+    indent += 4
+    print("{}wrappers:".format(" " * indent))
+
+    for wrapper in group.wrappers():
+        print("{}{}".format(" " * (indent + 4), wrapper))
+        print_with_indent(str(wrapper.value(False)), indent + 8)
+
+    print("{}groups:".format(" " * indent))
+
+    for subgroup in group.groups():
+        print_group(subgroup, indent + 4)
+
+def print_with_indent(msg, indent):
+    indent_str = " " * indent
+    print(indent_str + msg.replace("\n", "\n" + indent_str))
+
+def seismo_to_txt(seismo, filename):
+    with open(filename,  "w") as f:
+        for x in np.nditer(seismo):
+            f.write(f"{x}\n")
+        
+
+def test_WaveEquationAcousticSolverAdjoint():
+    run_adjoint_validation( "twoLayers2000x2000x2000.xml", "acousticSolver" )
+
+def run_adjoint_validation( xmlFile, solverName ):
+    argv = [ "testWaveEquationAdjoint.py", "-i", xmlFile,
+             "-z", str(nRank) ]
+    geosx=pygeosx.initialize(rank,argv)
+    solver=geosx.get_group(f"/Solvers/{solverName}")
+    srcPos=solver.get_wrapper("sourceCoordinates").value()
+    recvPos=solver.get_wrapper("receiverCoordinates").value()
+    event=geosx.get_group("/Events")
+    maxTime=event.get_wrapper("maxTime").value()
+    eventSolver=geosx.get_group("/Events/solverApplications")
+    dt=eventSolver.get_wrapper("forceDt").value()
+
+    vel_info=False
+    if vel_info: 
+        # get velocity infornation: min,max.
+        objectPath="Region" 
+        meshName="modelbicouche"
+        velocity =solver.get_wrapper("/domain/MeshBodies/"+meshName+"/meshLevels/Level0/ElementRegions/elementRegionsGroup/"+objectPath+"/elementSubRegions/hexahedra/mediumVelocity").value()
+        minVal=np.zeros(1)
+        maxVal=np.zeros(1)
+        minVal[0]=np.min(velocity.to_numpy())
+        maxVal[0]=np.max(velocity.to_numpy())
+        comm.Barrier()
+        comm.Allreduce(MPI.IN_PLACE, maxVal, op=MPI.MAX)
+        comm.Allreduce(MPI.IN_PLACE, minVal, op=MPI.MIN)
+        comm.Barrier()
+
+    if rank ==0 :
+       print()
+       print("===============================================================")
+       print(" Running pygeosx with solver:",solver)
+       print(" Source coordinatesi : ",srcPos.to_numpy())
+       print(" Receiver coordinates: ",recvPos.to_numpy())
+       print(" simulation duration : ",maxTime)
+       print(" Time Step           : ",dt)        
+       if vel_info:
+           print(" Min Vel             : ",minVal)
+           print(" Max Vel             : ",maxVal)
+       print("===============================================================")
```

Comment:
@TotoGaz and I have been trying to systematically replace any print calls with python logging calls.  This can greatly improve our ability to debug problems/outputs